roles/mirror_ocp_release/tasks/facts.yml

```diff
@@ -7,8 +7,9 @@
   when:
     - mor_version is version("4.8", ">=")
   ansible.builtin.shell: >
-    {{ mor_cache_dir }}/{{ mor_version }}/{{ mor_installer }} coreos print-stream-json >
-    {{ mor_cache_dir }}/{{ mor_version }}/rhcos.json
+    flock -x -w 30 "{{ mor_cache_dir }}/{{ mor_version }}/.installer.lock" -c
```

Comment:
@tonyskapunk, I accepted your commit but then I realized this lock and the one for the installer must be the same one, because in the end there is this interdependency.

If the openshift-install binary is being extracted, we have to wait in order to use it to export the rhcos.json file, thus the use of the same lock.

Reply:
I see, but how does this task identifies between waiting for another task running `mor_installer coreos print-stream.json` from the extraction of the installer?

I believe it should be separated, i.e. if a job has reached this task, then the `mor_installer ` must exist already, and anything attempting to re-write it should not thanks to the other file lock mechanism in installer.yml.

That's why I think this locking mechanism must be different and focused on avoiding multiple jobs writing the same rhcos.json., wydt?

Reply:
As we discussed online, the thing is that:

1) a job trying to extract the installer while there's another job already doing so will wait for the installer to be extracted and move on to the next task.
2) a job trying to generator the rhcos.json file while there's another job extracting the installer will wait for the installer to be extracted and then generate the rhcos.json file.
3) a job trying to extract the installer while there's another job generating the rhcos.json file (this means the installer is already extracted), will wait for the rhcos to be generated and then move on to the next task.
4) a job trying to generate the rhcos.json while other job is generating it, will wait for the first job to finish generating and then generate it itself. The resulting file will be the same one but this is acceptable.

So, if I'm not missing anything, we should be safe with this solution.